tests/e2e/specs/graph.cy.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+function waitForGraphLayout () {
+  // wait for the initial graph layout to be performed
+  cy
+    .get('.c-graph:first')
+    .then(($el) => { checkGraphLayoutPerformed($el) })
+}
+
+function checkGraphLayoutPerformed ($el, depth = 0) {
+  // Check if the graphID has been set (indicating successful layout) or wait
+  // one second.
+  // This is a recursive function which will be called up to 10 times. We can't
+  // do this in a for loop as otherwise the waits aren't chained correctly.
+  if (depth > 10) {
+    expect('graph loaded').to.equal(true)
```

Comment:
Unless I am missing something, this is `'graph loaded' === true` which will always fail. Presumably it's not being hit?

Reply:
Yep, this test is guaranteed to fail. It gets called if the graph doesn't load in time.

Reply:
Ah, I understand now. According to the chai docs you should be able to do
```suggestion
    expect.fail('graph did not load')
```